src/imageLoader/wadors/metaDataManager.js

```diff
@@ -11,7 +71,26 @@ function add(imageId, metadata) {
 function get(imageId) {
   const imageURI = imageIdToURI(imageId);
 
-  return metadataByImageURI[imageURI];
+  let metadata = metadataByImageURI[imageURI];
+
+  if (!metadata) {
+    // in this case try see if the imageId corresponds to multiframe
+    const imageIdFrameless = imageURI.slice(
```

Comment:
this doesn't make sense, you are assuming here that if there is no metadata then it is multiframe, you should do a multiframe check first.

Reply:
The dicomweb client returns only one imageid for an multiframe dicom serie. So only one imageid is registered in metadataByImageURI. If metadata returned is undefined, we only have two options. Or there is no metadata at all, or the client is requesting an imageid from a frame.